android/app/src/main/java/org/fossasia/openevent/core/schedule/DayScheduleFragment.java

```diff
@@ -134,39 +115,26 @@ public void onChanged() {
         dayScheduleAdapter.registerAdapterDataObserver(adapterDataObserver);
     }
 
-    public void filter(List<String> selectedTracks) {
-        if(dayScheduleAdapter == null)
-            return;
-
-        realmRepo.getSessionsByDate(date, SortOrder.sortTypeSchedule())
-                .addChangeListener((sortedSessions, orderedCollectionChangeSet) -> {
-                    sessions.clear();
-                    sessions.addAll(sortedSessions);
-
-                    filteredSessions.clear();
-                    if (selectedTracks.isEmpty()) {
-                        filteredSessions.addAll(sessions);
-                    } else {
-                        for(int i = 0; i < sessions.size() ; i++) {
-                            String trackName = sessions.get(i).getTrack().getName();
-                            if (selectedTracks.contains(trackName)) {
-                                filteredSessions.add(sessions.get(i));
-                            }
-                        }
-
-                        if(searchText!=null)
-                            dayScheduleAdapter.filter(searchText);
-                    }
-
-                    if(SortOrder.sortOrderSchedule() == SortOrder.SORT_ORDER_DESCENDING)
-                        Collections.reverse(filteredSessions);
-
-                    dayScheduleAdapter.notifyDataSetChanged();
-
-                    handleVisibility();
-                });
+    public void filterByTracks(List<String> selectedTracks) {
+        dayScheduleFragmentViewModel.getSortedSessionsByTracks(searchText, selectedTracks).observe(DayScheduleFragment.this, trackSessions -> {
+            filteredSessions.clear();
+            filteredSessions.addAll(trackSessions);
+            dayScheduleAdapter.notifyDataSetChanged();
+            handleVisibility();
+        });
+    }
+
+    public void filterByQuery(String query) {
+        searchText = query;
+        dayScheduleFragmentViewModel.getSessionsBySearchText(searchText).observe(DayScheduleFragment.this, matchingSessions -> {
+            filteredSessions.clear();
+            filteredSessions.addAll(matchingSessions);
+            dayScheduleAdapter.notifyDataSetChanged();
```

Comment:
@srv-twry I tried to use dayScheduleAdapter.animateTo(matchingSessions) for this but it lagged when I started typing in the search query. I'm not sure why the animateTo did not work well for this.

Reply:
Its okay now, I now use animateTo in onQueryTextChange instead.